docs/design/segments.md

```diff
@@ -23,231 +23,196 @@ title: "Segments"
-For example, if your real-time ingestion creates 3 segments that were sharded with linear shard spec, and only two of the segments were loaded in the system, queries would return results only for those 2 segments.
+    A file containing metadata (filenames and offsets) about the contents of the other `smoosh` files
 
-## Schema changes
+* `XXXXX.smoosh`
 
-## Replacing segments
+    A number of files containing concatenated binary data.
 
-Druid uniquely
-identifies segments using the datasource, interval, version, and partition number. The partition number is only visible in the segment id if
-there are multiple segments created for some granularity of time. For example, if you have hourly segments, but you
-have more data in an hour than a single segment can hold, you can create multiple segments for the same hour. These segments will share
-the same datasource, interval, and version, but have linearly increasing partition numbers.
+    The `smoosh` files represent multiple files "smooshed" together in order to minimize the number of file descriptors that must be open to house the data. They are files of up to 2GB in size (to match the limit of a memory mapped ByteBuffer in Java). The `smoosh` files house individual files for each of the columns in the data as well as an `index.drd` file with extra metadata about the segment.
 
-```
-foo_2015-01-01/2015-01-02_v1_0
-foo_2015-01-01/2015-01-02_v1_1
-foo_2015-01-01/2015-01-02_v1_2
-```
+Additionally, a column called `__time` refers to the time column of the segment.
 
-In the example segments above, the `dataSource = foo`, `interval = 2015-01-01/2015-01-02`, `version = v1`, and `partitionNum = 0`.
-If at some later point in time, you reindex the data with a new schema, the newly created segments will have a higher version id.
+In the codebase, segments have an internal format version. The current segment format version is `v9`.
 
-```
-foo_2015-01-01/2015-01-02_v2_0
-foo_2015-01-01/2015-01-02_v2_1
-foo_2015-01-01/2015-01-02_v2_2
-```
+## Implications of updating segments
 
-Druid batch indexing (either Hadoop-based or IndexTask-based) guarantees atomic updates on an interval-by-interval basis.
-In our example, until all `v2` segments for `2015-01-01/2015-01-02` are loaded in a Druid cluster, queries exclusively use `v1` segments.
-Once all `v2` segments are loaded and queryable, all queries ignore `v1` segments and switch to the `v2` segments.
-Shortly afterwards, the `v1` segments are unloaded from the cluster.
+Druid batch indexing (either Hadoop-based or IndexTask-based) guarantees atomic updates on an interval-by-interval basis. In our example, until all `v2` segments for `2015-01-01/2015-01-02` are loaded in a Druid cluster, queries exclusively use `v1` segments. Once all `v2` segments are loaded and queryable, all queries ignore `v1` segments and switch to the `v2` segments. Shortly afterwards, the `v1` segments are unloaded from the cluster.
```

Comment:
Observation. In the above paragraph, we talked about (current segment) version, and said it is "v9". Here, we're talking about (segment data) versions and say it is "v1" and "v2". I think the casual reader will be confused. Maybe:

> Druid uses versioning to manage updates to create a form of MVCC (multi-version concurrency control.) These MVCC versions are distinct from the segment format version discussed above.